lite2/client.go

```diff
@@ -926,53 +926,75 @@ func (c *Client) backwards(
 	return nil
 }
 
+type errNoRespFromWitness struct {
+	Reason       error
+	WitnessIndex int
+}
+
+func (e errNoRespFromWitness) Error() string {
+	return fmt.Sprintf("failed to get a header from witness: %v", e.Reason)
+}
+
+type errInvalidHeader struct {
```

Comment:
Do you want to move the errors to the error file. Also I believe we have a similar error already so maybe best to clearly distinguish them in the name